marketplace/src/benchmarking.rs

```diff
@@ -17,236 +17,166 @@
 	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().name, new_name);
+		let marketplace = Marketplaces::<T>::get(marketplace_id).unwrap();
+		assert_eq!(marketplace.commission_fee, Some(MarketplaceFee::Percentage(PERCENT_50)));
+		assert_eq!(marketplace.listing_fee, Some(MarketplaceFee::Percentage(PERCENT_50)));
+		assert_eq!(marketplace.account_list, Some(marketplace_account_list));
+		assert_eq!(marketplace.offchain_data, Some(marketplace_offchain_data));
 	}
 
 	set_marketplace_mint_fee {
-		prepare_benchmarks::<T>();
-
 		let old_mint_fee = Marketplace::<T>::marketplace_mint_fee();
-		let new_mint_fee = 1000u32;
-
+		let new_mint_fee = 20u32;
 	}: _(RawOrigin::Root, new_mint_fee.clone().into())
 	verify {
 		assert_ne!(old_mint_fee, new_mint_fee.clone().into());
 		assert_eq!(Marketplace::<T>::marketplace_mint_fee(), new_mint_fee.into());
 	}
 
-	set_marketplace_commission_fee {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
-		let commission_fee = 67;
-	}: _(get_origin::<T>("ALICE"), mkp_id, commission_fee)
-	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().commission_fee, commission_fee);
-	}
-
-	set_marketplace_uri {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
-		let uri = BoundedVec::try_from("test".as_bytes().to_vec()).unwrap();
-	}: _(get_origin::<T>("ALICE"), mkp_id, uri.clone())
-	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().uri, uri);
-	}
-
-	set_marketplace_logo_uri {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
-		let uri = BoundedVec::try_from("test".as_bytes().to_vec()).unwrap();
-	}: _(get_origin::<T>("ALICE"), mkp_id, uri.clone())
+	list_nft {
+		prepare_benchmarks::<T>();
+		let alice: T::AccountId = get_account::<T>("ALICE");
+		let alice_origin = origin::<T>("ALICE");
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
+		Marketplace::<T>::set_marketplace_configuration(
+			alice_origin.clone().into(),
+			marketplace_id,
+			ConfigOp::Set(MarketplaceFee::Percentage(PERCENT_50)),
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+		).unwrap();
+		let nft_id = T::NFTExt::create_nft(alice, BoundedVec::default(), PERCENT_50, None, false).unwrap();
+	}: _(alice_origin, nft_id, 10u32.into(), marketplace_id)
 	verify {
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().logo_uri, uri);
+		assert_eq!(T::NFTExt::get_nft(nft_id).unwrap().state.listed_for_sale, true);
+		assert!(Marketplace::<T>::nfts_for_sale(nft_id).is_some());
 	}
 
-	add_account_to_disallow_list {
-		let (mkp_id, ..) = prepare_benchmarks::<T>();
-
-		let bob: T::AccountId = get_account::<T>("BOB");
-		let bob_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(bob.clone());
-
-	}: _(get_origin::<T>("ALICE"), mkp_id, bob_lookup.into())
+	unlist_nft {
+		prepare_benchmarks::<T>();
+		let alice: T::AccountId = get_account::<T>("ALICE");
+		let alice_origin = origin::<T>("ALICE");
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
```

Comment:
You should probably use `next_marketplace_id` instead of `get_next_marketplace_id`